src/snapshot_tests/scene_test.rs

```diff
@@ -0,0 +1,192 @@
+use anyhow::Result;
+use compositor_common::{
+    scene::{InputId, NodeId, OutputId, Resolution, SceneSpec},
+    Frame,
+};
+use compositor_render::{FrameSet, Renderer};
+use std::{collections::HashSet, fmt::Display, fs, path::PathBuf, sync::Arc, time::Duration};
+
+use super::{
+    test_case::TestInput,
+    utils::{are_snapshots_equal, frame_to_rgba, SNAPSHOTS_DIR_NAME},
+};
+
+pub struct SceneTest {
+    pub test_name: &'static str,
+    pub scene: Arc<SceneSpec>,
+    pub inputs: Vec<TestInput>,
+    pub renderer: Renderer,
+    pub timestamps: Vec<Duration>,
+    pub outputs: Vec<&'static str>,
+}
+
+impl SceneTest {
+    pub fn run(&self) -> Result<()> {
+        let mut produced_outputs = HashSet::new();
+        let snapshots = self.generate_snapshots()?;
+        for new_snapshot in snapshots.iter() {
```

Comment:
```suggestion
        for snapshot in snapshots.iter() {
```
new_snapshot is misleading